source/server-discovery-and-monitoring/server-monitoring.rst

```diff
@@ -662,17 +662,12 @@ The event API here is assumed to be like the standard `Python Event
         heartbeatFrequencyMS = heartbeatFrequencyMS
         minHeartbeatFrequencyMS = 500
 
-        # Heartbeat command
-        if versionedApi or handshakeResponse.helloOk:
-            helloCommand = hello
-        else
-            helloCommand = legacy hello
-
         # Internal Monitor state:
         connection = Null
-        description = default ServerDescription
+        # server API versioning implies that the server supports hello
+        description = default ServerDescription(helloOk=versionedApi)
```

Comment:
To skip storing on the ServerDescription:
```suggestion
        description = default ServerDescription()
        helloOk = versionedApi != Null
```

Reply:
`helloOk` initialization/reset logic centralized in `setupConnection`.